src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -184,187 +184,248 @@ public async Task ReinstateSubscriptionAsync(Guid organizationId)
-        organization.UseSso = newPlan.HasSso;
-        organization.UseKeyConnector = newPlan.HasKeyConnector;
-        organization.UseScim = newPlan.HasScim;
-        organization.UseResetPassword = newPlan.HasResetPassword;
-        organization.SelfHost = newPlan.HasSelfHost;
-        organization.UsersGetPremium = newPlan.UsersGetPremium || upgrade.PremiumAccessAddon;
-        organization.UseCustomPermissions = newPlan.HasCustomPermissions;
-        organization.Plan = newPlan.Name;
+        organization.PlanType = passwordManagerPlan.Type;
+        organization.Seats = (short)(passwordManagerPlan.BaseSeats + upgrade.AdditionalSeats);
+        organization.MaxCollections = passwordManagerPlan.MaxCollections;
+        organization.UseGroups = passwordManagerPlan.HasGroups;
+        organization.UseDirectory = passwordManagerPlan.HasDirectory;
+        organization.UseEvents = passwordManagerPlan.HasEvents;
+        organization.UseTotp = passwordManagerPlan.HasTotp;
+        organization.Use2fa = passwordManagerPlan.Has2fa;
+        organization.UseApi = passwordManagerPlan.HasApi;
+        organization.SelfHost = passwordManagerPlan.HasSelfHost;
+        organization.UsePolicies = passwordManagerPlan.HasPolicies;
+        organization.MaxStorageGb = !passwordManagerPlan.BaseStorageGb.HasValue ?
+            (short?)null : (short)(passwordManagerPlan.BaseStorageGb.Value + upgrade.AdditionalStorageGb);
+        organization.UseGroups = passwordManagerPlan.HasGroups;
+        organization.UseDirectory = passwordManagerPlan.HasDirectory;
+        organization.UseEvents = passwordManagerPlan.HasEvents;
+        organization.UseTotp = passwordManagerPlan.HasTotp;
+        organization.Use2fa = passwordManagerPlan.Has2fa;
+        organization.UseApi = passwordManagerPlan.HasApi;
+        organization.UseSso = passwordManagerPlan.HasSso;
+        organization.UseKeyConnector = passwordManagerPlan.HasKeyConnector;
+        organization.UseScim = passwordManagerPlan.HasScim;
+        organization.UseResetPassword = passwordManagerPlan.HasResetPassword;
+        organization.SelfHost = passwordManagerPlan.HasSelfHost;
+        organization.UsersGetPremium = passwordManagerPlan.UsersGetPremium || upgrade.PremiumAccessAddon;
+        organization.UseCustomPermissions = passwordManagerPlan.HasCustomPermissions;
+        organization.Plan = passwordManagerPlan.Name;
         organization.Enabled = success;
         organization.PublicKey = upgrade.PublicKey;
         organization.PrivateKey = upgrade.PrivateKey;
+        organization.UsePasswordManager = true;
+
+        if (secretsManagerPlan != null && useSecretsManager)
+        {
+            organization.SmSeats = (short)(secretsManagerPlan.BaseSeats + upgrade.AdditionalSmSeats);
+            organization.SmServiceAccounts = (int)(secretsManagerPlan.BaseServiceAccount + upgrade.AdditionalServiceAccount);
+            organization.UseSecretsManager = true;
```

Comment:
I think we should set this either way. (i.e. move it outside of this `if` block)

Reply:
resolved